scripts/e2e/setup-cluster.sh

```diff
@@ -103,6 +103,12 @@ install_knative() {
     kubectl rollout status deployment/domainmapping-webhook -n knative-serving -w --timeout=${TIMEOUT}
     kubectl rollout status deployment/webhook -n knative-serving -w --timeout=${TIMEOUT}
 
+    # Update knative config-deployment config map to allow resolving of the local e2e image repository
+    kubectl get configmap -n knative-serving config-deployment -o yaml > temp.yaml
+    yq -i '.data.registries-skipping-tag-resolving |= ("\(.),${DOCKER_REGISTRY}" | envsubst)' temp.yaml
+    kubectl apply -f temp.yaml
+    kubectl rollout restart deployment -n knative-serving controller
```

Comment:
These additional steps are needed to ensure Knative does not perform additional steps to resolve the local e2e image registry (the image already exists but Knative attempts to connect to the registry as an extra 
step before it pulls any image from it). This error shows up when that happens https://github.com/caraml-dev/merlin/actions/runs/11861830172/job/33060193167#step:17:7853. See https://github.com/knative/serving/issues/14117 and https://stackoverflow.com/questions/73619481/unable-to-pull-from-knative-but-able-from-a-normal-deploy for more details.